File: todo-angular-frontend/src/app/shared/entity/TodoEntry.ts
export class TodoEntry {
  id: number;
  description: string;
  createdAt: Date;
  updatedAt: Date;
  dueDate: Date;
  state: string;
  important: boolean;
  userId: number;

  constructor(id: number, description: string, createdAt: Date, updatedAt: Date, dueDate: Date,
              state: string, important: boolean , userId: number) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.description = description;
    this.dueDate = dueDate;
    this.state = state;
    this.userId = userId;
  }
}

File: todos-springboot-backend/src/main/java/de/webapp/springboot/todos/todo/domain/TodoEntry.java
package de.webapp.springboot.todos.todo.domain;

import de.webapp.springboot.todos.user.domain.User;
import lombok.*;
import lombok.experimental.Wither;

import javax.persistence.*;
import java.time.LocalDateTime;

@Table(name = "TODO_ENTRY")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString(exclude = "user")
@Builder
@Wither
@Entity
public class TodoEntry {

    @Id
    @GeneratedValue
    @Column(name = "ID")
    private long id;

    @Column(name = "DESCRIPTION")
    private String description;

    @Column(name = "CREATED_AT")
    private LocalDateTime createdAt;

    @Column(name = "UPDATED_AT")
    private LocalDateTime updatedAt;

    @Column(name = "DUE_DATE")
    private LocalDateTime dueDate;

    @Column(name = "STATE")
    @Enumerated
    private TodoEntryState state;

    @Column(name = "IMPORTANT")
    private boolean important;

    @ManyToOne
    @JoinColumn(name = "user")
    private User user;

}
